### PR TITLE
2176 Ensembl API for both MANE and Canonical transcript title

### LINKED ISSUE #1868: Use MANE transcripts for Variant titles
Currently Variant titles are derived via the following formula:
1. ClinVar preferred title
2. If no 1. then title based the single canonical RefSeq transcript from Ensembl VEP
3. If no 2. then title based on GRCh38 genomic HGVS

This would change to incorporate the MANE transcripts:
1. ClinVar preferred title
2. If no 1. then title based on the RefSeq transcript MANE from CAR
2. If no 2. then title based the single canonical RefSeq transcript from Ensembl VEP
3. If no 3. then title based on GRCh38 genomic HGVS

### PR DESCRIPTION
**Pull request by shaungc-su**
#2176 

## This ticket does
- Change data source from LDH + Genomic Allele Registry to Ensembl-HGVS-VEP API to enhance availability of MANE transcripts, since Ensembl API is the original source of MANE info.
- Rewrite canonical transcript title generating method to more accurately select a desired transcript, and integrate with MANE to share Ensembl API data.
- Remove unnecessary title fetching in #1868  (PR: #2159 ) if Clinvar Variant Title available to improved response time, since Ensembl API could be slow (5+ sec)
- Auto focus add variant modal input to save user clicks
- Bug fix #1868 when MANE is not available the modal UI hangs.

## Quick curator test steps 
([Test Instance Link](https://2176-sc-mane-ensembl-rev05.demo.clinicalgenome.org/)):
- Go to a GCI classification page, and add individual / group / ... evidence
- In the evidence form, Associated Variant section, add & save a variant to the evidence
- The variant title is shown in Associated Variant section. Depend on which variant title is used, the label is displayed as either one of the following: ClinVar Preferred Title / MANE Transcript HGVS Title / Canonical Transcript HGVS Title / Genomic HGVS Title

Sample variants to test for title using MANE: CA501068, CA501078
Sample variants to test for title using ClinVar title: CA321211, CA015944, and any variants in ClinVar
Sample variants to test for title using Canonical: can't find one so did not test

### CODE REVIEW

File: src/clincoded/static/libs/parse-resources.js [hunk 1]
```diff
@@ -504,47 +520,119 @@ function parseCarHgvsHandler(hgvs_temp, variant) {
...
+        if (!hgvsValue) {
+            continue;
+        }
+        
+        const [transcriptId, nucleotideChange] = hgvsValue.split(':')
+        const {
+            proteinEffect: {
+                hgvs = ''
+            } = {}
+        }= transcript;
```
Comment (markmandell) [inline src/clincoded/static/libs/parse-resources.js:623]: spacing between `}` and `=` on line 623 👍

File: src/clincoded/static/libs/get_canonical_transcript.js [hunk 2]
```diff
@@ -1,16 +1,153 @@
...
+    }
+
+    // NM Singularity Test
+
+    const canonicalTranscriptsStartByNM = canonicalTranscriptsFromEnsembl.filter(({ hgvsc }) => (typeof hgvsc === 'string' && hgvsc.trim().startsWith('NM')))
+
+    if (canonicalTranscriptsStartByNM.length === 1) {
+        return extended ? canonicalTranscriptsStartByNM[0] : canonicalTranscriptsStartByNM[0].hgvsc;
+    }
+
```
[the remark this answers is not in the document]
Comment (shaungc-su) [reply src/clincoded/static/libs/get_canonical_transcript.js:70]: This is a really nice catch among the complicate logics.. The code is actually wrong, the comment is right after looking at the 4th example provided by Matt written in func head comments. 

I'll fix the case where there're multi-NMs, we should return `null` right away. Thanks!

File: src/clincoded/static/components/add_external_resource.js [hunk 3]
```diff
@@ -600,26 +600,43 @@ function clinvarQueryResource() {
...
+                                return res(data);
+                            })
+                    }
+
+                    return res(data);
+                })
+                .then((data) => {
+                    // since we have parseClinvar parse extended data,
+                    // we need to remove the additional fields that are not recognized by backend
+                    console.log('additonal keys', extendedVariantKeysAdded);
```
[the remark this answers is not in the document]
Comment (shaungc-su) [reply src/clincoded/static/components/add_external_resource.js:633]: Good catch! Gotta remove these logs.

File: src/clincoded/static/libs/get_canonical_transcript.js [hunk 4]
```diff
@@ -1,16 +1,153 @@
...
+                if (item.hgvs && item.hgvs.length) {
+                    for (const transcript of item.hgvs) {
+                        if (transcript === canonicalTranscript) {
+                            let proteinChange;
+                            const transcriptStart = transcript.split(':')[0];
+                            const transcriptEnd = transcript.split(':')[1];
+                            if (item.proteinEffect && item.proteinEffect.hgvs) {
+                                proteinChange = item.proteinEffect.hgvs.split(':')[1];
+                            }
+                            return `${transcriptStart}(${item.geneSymbol}):${transcriptEnd}${proteinChange ? ` (${proteinChange})` : ''}`;
```
[the remark this answers is not in the document]
Comment (shaungc-su) [reply src/clincoded/static/libs/get_canonical_transcript.js:124]: Good idea, will add checks for them.

**Pull request merged**